### ISSUE TITLE
macOS: black screen & flicker with OpenGL display driver

### ISSUE THREAD

**Issue by CarterLi** [mgba-emu/mgba#1619]
It works with OpenGL 1.x
0.8-6018-d4e30b74

And another issue: I got a white rectangle after exiting fullscreen:

![image](https://user-images.githubusercontent.com/6134068/72401886-576c0b80-3788-11ea-8c9b-097b71f1a2f1.png)

**Comment by CarterLi** [comment]
Verified white screen issue worked on 0.7.3

**Comment by endrift** [comment]
The fullscreen issue is #1212. 
What's your GPU and macOS version?

**Comment by CarterLi** [comment]
Catalina 10.15.2 (19C57)
Intel Iris Pro 1536 MB

**Comment by endrift** [comment]
Were you using the OpenGL enhancement renderer too? That was broken on macOS until 9a0da39848a540ef3909f0051160843f093dfd83, which I just pushed a few minutes ago.

**Comment by CarterLi** [comment]
I'll test it after the nightly build comes out

**Comment by endrift** [comment]
It's already up: https://s3.amazonaws.com/mgba/build/mGBA-build-2020-01-20-osx-6030-9a0da39848a540ef3909f0051160843f093dfd83.tar.xz

**Comment by CarterLi** [comment]
The black screen issue disappear, but the graph size doesn't seem to be right

![image](https://user-images.githubusercontent.com/6134068/72692350-5bbc6e00-3b66-11ea-9ed2-9ecc728a02e9.png)

**Comment by endrift** [comment]
What do you have the enhancement scale set to? And do you have integer scaling enabled?

**Comment by CarterLi** [comment]
> What do you have the enhancement scale set to?

3x

> And do you have integer scaling enabled?

yes

![image](https://user-images.githubusercontent.com/6134068/72692455-003eb000-3b67-11ea-87c3-b2ac34908446.png)

The graph size get right if I reset the frame size, but the issue happens when I restart mgba

**Comment by endrift** [comment]
I'm not sure I understand the issue. Can you make a screen recording of it?

**Comment by kode54** [comment]
That appears to be clusters of 3x3 blocks of native pixels. Which have the effective resolution of 1.5x on an @2xHiDPI macOS display. You'll need to double your scaling as long as you're running it on a 2x display. Or the emulator will need to account for the native scaling of whichever display it's dropped onto, which can change any time it's moved across screens.

**Comment by endrift** [comment]
mGBA is supposed to handle that that already. If it's not then something weird is going on, because I test on a HiDPI screen already.

**Comment by kode54** [comment]
Wait, so you automatically scale the OpenGL scale option according to the DPI of the current display?

**Comment by endrift** [comment]
No, I scale the generated texture by the current DPI. If you set the scale to 3x and the frame size to 3x you should wind up with a 3x texture rendered to a 18x window--3 x 3 x 2.
